### PR TITLE
Claims / Sign in page should redirect to persona select

### PR DESCRIPTION
**Pull request by Nitemaeric**
## Context

We want to lay down the foundations of user pages and flows. There is no data added, just _blank_ pages to highlight routing structures.

## Changes proposed in this pull request

- Remove "University" persona from Claims personas
- Redirect Claims users to sign-in page when landing on service root url
- Add "account" and "sign out" navigation items to navbar when signed in

## Guidance to review

- Head to the Claims service
- Be redirected to the personas page
- Login as any of the personas
- Be redirected to their internal pages (General users pending)

## Link to Trello card

https://trello.com/c/ow40pfAm/75-update-landing-page-of-app-to-sign-in-or-persona-select

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated or [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

## Screenshots / Videos

https://github.com/DFE-Digital/itt-mentor-services/assets/4231530/eb2fd6d8-ea62-4f6b-b7c7-90aef4f3016e

### CODE REVIEW

File: app/controllers/claims/pages_controller.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class Claims::PagesController < ApplicationController
+  before_action :authenticate_user!
```
Comment (CatalinVoineag) [inline app/controllers/claims/pages_controller.rb:2]: Should we just put this in the ApplicationController? And if any controllers don't need it we can skip it in there.

File: app/controllers/claims/schools_controller.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+class Claims::SchoolsController < ApplicationController
```
Comment (CatalinVoineag) [inline app/controllers/claims/schools_controller.rb:1]: Related to the comment above, don't we need to auth the user here as well?

File: app/controllers/personas_controller.rb [hunk 3]
```diff
@@ -2,6 +2,6 @@
 
 class PersonasController < ApplicationController
   def index
-    @personas = Persona.all.decorate
+    @personas = Persona.where(service: current_service).decorate
```
Comment (elceebee) [inline app/controllers/personas_controller.rb:5]: You can use the scope on the model for each service.

```suggestion
    @personas = Persona.public_send(current_service).decorate
```

File: app/controllers/claims/pages_controller.rb (hunk 1 above)
Comment (elceebee) [reply app/controllers/claims/pages_controller.rb:2]: I agree with Catalin that this would be better on the ApplicationController and we skip it when not needed. It is easier to require auth by default since that is what we will need most often.

**Review by elceebee — APPROVED** [review]
Would be good to merge [this PR](https://github.com/DFE-Digital/itt-mentor-services/pull/40). it will fix the broken tests and adds a few more tests for returning the correct user.

I agree with Catalin about requiring authentication by default and skipping as an exception rather than the other way around.

File: app/controllers/claims/pages_controller.rb (hunk 1 above)
Comment (Nitemaeric) [reply app/controllers/claims/pages_controller.rb:2]: I think I'd propose something slightly different.

```
class Claims::PagesController < Claims::Base::ApplicationController # Base / Users / Main any of these namespaces work for me
end

class Claims::Base::ApplicationController < ApplicationController
  before_action :authenticate_user!
end
```

This way, we have a base controller for general user routes and one for support. We also have the top-level `ApplicationController`'s action hooks unmodified and likely the path of least surprise. Anything inheriting from `ApplicationController` should have access to all helpers and custom controller methods, but without any additional behaviour added.

What do you both think?

Comment (Nitemaeric) [reply app/controllers/claims/pages_controller.rb:2]: I think auth by default is also viable now that I think about it though 🤔 .

Comment (Nitemaeric) [reply app/controllers/claims/pages_controller.rb:2]: Yeah, scratch the above suggestion. Keep it simple.

Comment (CatalinVoineag) [reply app/controllers/claims/pages_controller.rb:2]: I guess I don't really see the value of having ` Claims::Base::ApplicationController` at the moment, feels like we are abstracting too soon.

I like the idea of auth by default and if you need to skip it for the support controller you can skip it or override the method from the `ApplicationController`.

In the end, I don't really have a strong opinion on this. Happy with both approaches

File: app/controllers/claims/support/application_controller.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class Claims::Support::ApplicationController < ApplicationController
+  before_action :authenticate_support_user!
```
Comment (CatalinVoineag) [inline app/controllers/claims/support/application_controller.rb:2]: Don't we need `skip_before_action :authenticate_user!` here? I think it will go through the application controller twice now.

We could also just override the `authenticate_user` method in here if you want.

**Review by CatalinVoineag — COMMENTED** [review]
Pulled this on local, works as expected 👍🏻 Just one comment that I think we should address.

File: app/controllers/claims/support/application_controller.rb (hunk 4 above)
Comment (ollietreend) [reply app/controllers/claims/support/application_controller.rb:2]: The structure of the equivalent `Placements::Support::ApplicationController` class changed before PR #42 was merged, so you'll probably want to update this file to match.

https://github.com/DFE-Digital/itt-mentor-services/blob/a485beef8f5ddedbf32f3484d749162214958fed/app/controllers/placements/support/application_controller.rb#L1-L11

In addition to that, both `[SERVICE]::Support::ApplicationController` classes will now need to update the `before_action` line so it doesn't re-declare `:authenticate_user!` since that's now in the base `ApplicationController` class.

In other words, both classes should now look like this:

```diff
class [SERVICE]::Support::ApplicationController < ApplicationController
-  before_action :authenticate_user!, :authorize_user!
+  before_action :authorize_user!

  private

  def authorize_user!
    return if current_user.support_user?

    redirect_to placements_root_path, alert: "You cannot perform this action"
  end
end
```

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -63,6 +66,18 @@
...
+    host! ENV['CLAIMS_HOST']
+    example.run
+    host! nil
+  end
+
+  config.around(:each, type: :system) do |example|
+    Capybara.app_host = "http://#{ENV['CLAIMS_HOST']}"
+    example.run
+    Capybara.app_host = nil
+  end
```
Comment (ollietreend) [inline spec/rails_helper.rb:80]: It looks like there's duplication here between this config and the config manually applied on system specs in @CatalinVoineag's PR.

https://github.com/DFE-Digital/itt-mentor-services/blob/1f7ff126fafb8160301460e2fba88a2d49720da6/spec/system/claims/view_organisations_spec.rb#L4-L8

Also: how will this config know which hostname to use – `CLAIMS_HOST` or `PLACEMENTS_HOST`?

File: spec/requests/sessions_spec.rb [hunk 6]
```diff
@@ -1,27 +1,23 @@
...
-    host! ENV["PLACEMENTS_HOST"]
-    example.run
-    host! nil
-  end
-
   describe "POST /auth/developer/callback" do
     it "returns http success" do
-      placements_user = create(:placements_user)
-      post auth_developer_callback_path,
+      claims_user = create(:claims_user)
```
Comment (ollietreend) [inline spec/requests/sessions_spec.rb:6]: Any reason why this was changed from `placements_user` to `claims_user`? Was this a meaningful change?

**Review by ollietreend — DISMISSED** [review]
This is looking good ✨

I've added one change request, a comment, and a question.

File: spec/support/capybara.rb
Comment (ollietreend) [inline spec/support/capybara.rb:1]: Does this branch need rebasing on `main` now that #51 has been merged in, or is this introducing duplicate config?

File: spec/rails_helper.rb (hunk 5 above)
Comment (Nitemaeric) [reply spec/rails_helper.rb:83]: This is the default, it sets the host to `CLAIMS` by default. I've made the change as I saw a previous PR sets `CLAIMS` to be the default host.

For system tests, I've added the `given_i_am_on_the_xxxxx_site` helper method, which is the only time we should really need to think about the site we're on. In unit tests, we shouldn't need to think about it.

Alternatively, we _could_ pass in a `service: :claims` or `service: :placements` in the spec describe options,

```ruby
RSpec.describe "Home Page", type: :system, service: :claims do
```

File: spec/requests/sessions_spec.rb (hunk 6 above)
Comment (Nitemaeric) [reply spec/requests/sessions_spec.rb:6]: Changing the default service to `:claims` means the service will look up a `Claims::User` by default.

**Comment by Nitemaeric** [comment]
[Update landing page of app to Sign In or Persona select](https://trello.com/c/ow40pfAm/75-update-landing-page-of-app-to-sign-in-or-persona-select)

**Review by CatalinVoineag — APPROVED** [review]
Tested this on local, works as expected. Good work
